ejb/src/test/java/biz/karms/sinkit/ejb/impl/BlacklistCacheServiceEJBTest.java

```diff
@@ -0,0 +1,222 @@
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Tomas Kozel
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class BlacklistCacheServiceEJBTest {
+
+    IoCRecord ioc1;
+    IoCRecord ioc2;
+    String key;
+    GsonBuilder gsonBuilder;
+    @Mock
+    private Logger log;
+
+    @Mock
+     RemoteCache<String, BlacklistedRecord> blacklistCache;
+
+    @InjectMocks
+    private BlacklistCacheServiceEJB blacklistCacheService;
+
+    @Before
+    public void setUp () throws Exception{
+        //ioc1, ioc2 and their corresponding key (both have the same)
+        //will be used throughout
+       ioc1 = new GsonBuilder().setDateFormat(IoCRecord.DATE_FORMAT)
+                .create().fromJson(new InputStreamReader(new FileInputStream("src/test/resources/ioc1.json")),
+                        IoCRecord.class);
+        key = DigestUtils.md5Hex(ioc1.getSource().getId().getValue());
+        ioc2 = new GsonBuilder().setDateFormat(IoCRecord.DATE_FORMAT)
+                .create().fromJson(new InputStreamReader(new FileInputStream("src/test/resources/ioc2.json")),
+                        IoCRecord.class);
+
+        JsonDeserializer<Calendar> calendarJsonDeserializer = (json, typeOfT, context) -> {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            Calendar c = Calendar.getInstance();
+            try {
+                Date date = sdf.parse(json.getAsJsonPrimitive().getAsString());
+                c.setTime(date);
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+            return c;
+        };
+        JsonDeserializer<Pair> pairJsonDeserializer = (json, typeOfT, context) -> {
+            Set<Map.Entry<String, JsonElement>> entrySet = json.getAsJsonObject().entrySet();
+            List<Pair<String, String>> list = entrySet.stream().map(entry -> new ImmutablePair<>(entry.getKey(), entry.getValue().getAsString())).collect(
+                    Collectors.toList());
+
+            return !list.isEmpty() ? list.get(0) : null;
+        };
+
+        gsonBuilder = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.IDENTITY).setDateFormat("yyyy-MM-dd")
```

Comment:
@hugeox What does it test here?

Reply:
This is not a test, it is for me to be able to parse the jsons using gson.

Reply:
Same for above